pkg/reconciler/ingress/ingress.go

```diff
@@ -623,3 +666,120 @@ func isRouteReady(r *routev1.Route) bool {
 	}
 	return false
 }
+
+func (r *Reconciler) reconcileSmmr(ctx context.Context, ing *v1alpha1.Ingress) error {
+	logger := logging.FromContext(ctx)
+
+	if err := r.reconcileNetworkPolicy(ctx, ing, false); err != nil {
+		return err
+	}
+
+	// update ServiceMeshMemberRole with the namespace info where knative routes created
+	smmr, err := r.smmrLister.ServiceMeshMemberRolls(smmrNamespace).Get(smmrName)
+	if err != nil {
+		return fmt.Errorf("failed to get SMMR in %q: %w", smmrNamespace, err)
+	}
+	newMembers, changed := appendIfAbsent(smmr.Spec.Members, ing.GetNamespace())
+	smmr.Spec.Members = newMembers
```

Comment:
This modifies the informers copy (from the lister call above). We should DeepCopy it and only do that inside the `changed` bracket.